papaya-core/src/main/java/pl/edu/agh/papaya/service/sprint/SprintService.java

```diff
@@ -94,27 +95,58 @@ Sprint createSprint(Sprint sprint) {
         return sprintRepository.findFirstByProjectIdOrderByDurationPeriodStartDesc(projectId);
     }
 
-    public Sprint closeSprint(Sprint sprint, Duration timeBurned, Duration timePlanned, LocalDateTime dateClosed) {
-        sprint.setTimeBurned(timeBurned);
-        sprint.setTimePlanned(timePlanned);
+    public Sprint closeSprint(Sprint sprint, Duration timeBurned, Duration finalTimePlanned, LocalDateTime dateClosed) {
         sprint.setDateClosed(dateClosed);
+        updateSprintStats(sprint, finalTimePlanned, timeBurned);
 
         sprintRepository.save(sprint);
 
         return sprint;
     }
 
-    public Optional<Sprint> getById(Long id) {
-        return sprintRepository.findById(id);
+    public void updateSprintStats(Sprint sprint, Duration finalTimePlanned, Duration timeBurned) {
+        SprintStats stats = sprint.getStats();
+
+        stats.setFinalTimePlanned(finalTimePlanned);
+        stats.setTimeBurned(timeBurned);
+
+        Duration totalDeclaredTime = getTotalDeclaredTimeBySprint(sprint);
+        stats.updateCoefficient(totalDeclaredTime);
+
+        stats.setAverageCoefficientCache(computeAvarageSprintCoefficient(sprint));
+    }
+
+    private Double computeAvarageSprintCoefficient(Sprint sprint) {
+        // date after start of this sprint because this one also has to be included
+        Double coefficient = sprintRepository.findAverageSprintCoefficientUpToDate(sprint.getDurationPeriod().getEnd());
```

Comment:
Nie wiem, czy bym tego nie zrobiła po datach początków z `<=`, ale  okej

Reply:
miałoby to sens